api/handlers/authentication.go

```diff
@@ -254,3 +257,145 @@ func sendOTP(userInfo string) {
+
+func (h Auth) ResetPassword(c *gin.Context) {
+	defer c.Request.Body.Close()
+	
+	//check the body of the request
+
+	var body models.SendResetRequest;
+	if err := c.BindJSON(&body); err != nil {
+		c.JSON(http.StatusBadRequest, models.Response{Error: "Invalid Request"})
+		return
+	}
+
+	//check if the email exists in the database
+
+	var user models.User;
+	h.DB.Where("email = ?", body.Email).First(&user);
+	if !h.checkUserExists(user.Email) {
+		c.JSON(http.StatusNotFound, models.Response{Error: "User does not exist"})
+		return
+	}
+
+
+	//generate a jwt token with the user's email
+	tempUser:= models.User{
+		Email: user.Email,
+	}
+
+	jwt, err := middleware.GenerateJWT(tempUser)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, models.Response{Error: "Error generating token"})
+		return
+	}
+
+	//send an email to the user with a temporary link to reset the password
+	linkURL := "http://localhost:8080/reset-password/" + jwt
+	email := models.Email{
+		From:    os.Getenv("COMPANY_EMAIL"),
+		To:      user.Email,
+		Subject: "Password Reset",
+		Body:    "Click here to reset your password: " + linkURL,
+	}
+	log.Println(email)
+	if err := sendMail(email); err != nil {
+		c.JSON(http.StatusInternalServerError, models.Response{Error: "Error sending email"})
+		return
+	}
+
+	c.JSON(http.StatusOK, models.Response{Data: "Email sent successfully"})
+
+}
+
+//activate reset password
+// @Summary Activate reset password
+// @Description Activate reset password
+// @Tags auth
+// @Accept json
+// @Produce json
+// @Param token path string true "Token"
+// @Success 200 {object} models.Response "Password reset link activated"
+// @Failure 400 {object} models.Response "Invalid Request"
+// @Router /auth/reset-password/reset [post]
+func (h Auth) ActivateResetPassword(c *gin.Context) {
+	defer c.Request.Body.Close()
+
+	//get the token from the url
+	authHeader := c.GetHeader("Authorization")
+	if authHeader == "" {
+		c.JSON(http.StatusBadRequest, models.Response{Error: "missing token"})
+		return
+	}
+	claims := middleware.GetClaims(c)
+	if claims == nil {
+		c.JSON(http.StatusBadRequest, models.Response{Error: "missing token"})
+		return
+	}
+
+	//check the body of the request
+
+	var body models.ResetPassword;
+	if err := c.BindJSON(&body); err != nil {
+		c.JSON(http.StatusBadRequest, models.Response{Error: "Invalid Request"})
+		return
+	}
+
+	//get user form the database
+	var user models.User;
+	h.DB.Where("email = ?", claims.Email).First(&user);
+	if !h.checkUserExists(user.Email) {
+		c.JSON(http.StatusNotFound, models.Response{Error: "User does not exist"})
+		return
+	}
+
+	//hash the new password
+	hashedPassword, err := hashPassword(body.NewPassword, &user)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, models.Response{Error: "Error hashing password"})
+		return
+	}
+
+	//update the user's password
+	h.DB.Model(&user).Updates(map[string]interface{}{
+		"password_hash": hashedPassword,
+		"salt":          user.Salt,
```

Comment:
Just a thought, but could using the same salt be problematic? Don't know if that's the case; just a hunch